specs/bellatrix/p2p-interface.md

```diff
@@ -85,13 +86,24 @@ The *type* of the payload of this topic changes to the (modified) `SignedBeaconB
 Specifically, this type changes with the addition of `execution_payload` to the inner `BeaconBlockBody`.
 See Bellatrix [state transition document](./beacon-chain.md#beaconblockbody) for further details.
 
+Blocks with execution enabled will be permitted to propagate regardless of the
+validity of the execution payload. This prevents network segregation between
+[optimistic](/sync/optimistic.md) and non-optimistic nodes.
+
 In addition to the gossip validations for this topic from prior specifications,
 the following validations MUST pass before forwarding the `signed_beacon_block` on the network.
 Alias `block = signed_beacon_block.message`, `execution_payload = block.body.execution_payload`.
 - If the execution is enabled for the block -- i.e. `is_execution_enabled(state, block.body)`
   then validate the following:
   - _[REJECT]_ The block's execution payload timestamp is correct with respect to the slot
     -- i.e. `execution_payload.timestamp == compute_timestamp_at_slot(state, block.slot)`.
+  - [REJECT] The block's parent (defined by `block.parent_root`) passes all
+      validation, excluding verification of the `block.body.execution_payload`.
+  - [IGNORE] The block's parent (defined by `block.parent_root`) passes all
+      validation, including verification of the `block.body.execution_payload`.
```

Comment:
Would it be reasonable to phrase this as:
```suggestion
  - [IGNORE] The block's parent (defined by `block.parent_root`) passes 
        verification of the `block.body.execution_payload`.
```

Given the reject rule above we already know that all verification except for the `execution_payload` passes so are just choosing between ignore and accept based on whether the execution payload is valid or not.

Although both versions still feel ambiguous about whether the result is ACCEPT or IGNORE in the case of the parent execution payload not yet being executed because the EL is syncing. I think it should be ACCEPT because the whole point is that an optimistic node can't validate the payload but also needs to avoid censoring blocks just because it can't validate the parent.

Reply:
I struggled to figure out the wording for this. The language for these propagation conditions is tricky!

I've applied @djrtwo's suggestion in b1ec9bcf.

I was tempted to go off piste and use a different specification language for these conditions, since it would have been easier to parse (IMO). However, that breaks standard and might get a bit unruly if we have *another* spec that needs to override some of these rules.

I'm still open to suggestion here :sweat_smile:

Reply:
yeah, I think I likely made a bad decision quite a while ago on the choice of `if not positive_condition: REJECT/IGNORE`. Might have been saner as `if negative_condition: REJECT/IGNORE`...